datamodel/include/data_types/rendering/RenderingContext.hpp

```diff
@@ -14,7 +14,7 @@
 
 namespace rcbe::rendering
 {
-struct RenderingContext {
+struct RenderingContext : std::enable_shared_from_this<RenderingContext> {
```

Comment:
It is a useful addition indded, but if it is not used, it is excessive. As of now